helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -139,6 +155,28 @@ protected void dumpDebugInfo() {
     }
   }
 
+  private void refreshCustomizedStateConfig(final HelixDataAccessor accessor) {
+    if (_propertyDataChangedMap.get(HelixConstants.ChangeType.CUSTOMIZED_STATE_CONFIG)
+        .getAndSet(false)) {
+      CustomizedStateConfig customizedStateConfig =
+          accessor.getProperty(accessor.keyBuilder().customizedStateConfig());
+      if (customizedStateConfig != null) {
+        _aggregationEnabledTypes =
+            new HashSet<>(customizedStateConfig.getAggregationEnabledTypes());
+      } else {
+        _aggregationEnabledTypes.clear();
+      }
+      _customizedStateCache.setAggregationEnabledTypes(_aggregationEnabledTypes);
```

Comment:
Can we explicitly call this outside the refreshCustomizedStateConfig and just before _customizedStateCache.refresh()? Since _customizedStateCache.setAggregationEnabledTypes and refresh must be ordered, putting them together is easier for us to understand and maintain.